### PR TITLE
add hs.grid.getCell to get the geometry rect of a grid cell

### PR DESCRIPTION
**Pull request by jfchevrette**
Implement hs.grid.getCell which returns the hs.geometry rect for the cell

### CODE REVIEW

**Comment by cmsj** [comment]
Thanks. I think the docstring could do with some clarification - it's talking about "the window", but if I understand correctly, this isn't typically being used in conjunction with a particular window, it's just getting the rect for an arbitrary cell on a screen, right?

**Comment by jfchevrette** [comment]
You are right. I updated the docstring in various places to remove refenrences to windows and also change the screen parameter to remove the (optional) tag.

File: extensions/grid/init.lua [hunk 1]
```diff
@@ -222,6 +222,35 @@ function grid.get(win)
...
+---
+--- Parameters:
+---  * cell - a cell object, i.e. an `hs.geometry` rect or argument to construct one
+---  * screen - an `hs.screen` object or argument to `hs.screen.find()` where the cell is located
+---
+--- Returns:
+---  * the `hs.geometry` rect for a cell on a particular screen
+function grid.getCell(cell, scr)
+  scr=screen.find(scr)
+  if not scr then log.e('screen cannot be nil') return grid end
```
Comment (lowne) [inline extensions/grid/init.lua]: better `return end`

File: extensions/grid/init.lua [hunk 2]
```diff
@@ -222,6 +222,35 @@ function grid.get(win)
...
+  cell=geom.new(cell)
+  local screenrect = scr:frame()
+  local screengrid = getGrid(scr)
+  -- sanitize, because why not
+  cell.x=max(0,min(cell.x,screengrid.w-1)) cell.y=max(0,min(cell.y,screengrid.h-1))
+  cell.w=max(1,min(cell.w,screengrid.w-cell.x)) cell.h=max(1,min(cell.h,screengrid.h-cell.y))
+  local cellw, cellh = screenrect.w/screengrid.w, screenrect.h/screengrid.h
+  local newframe = {
+    x = (cell.x * cellw) + screenrect.x + margins.w,
```
Comment (lowne) [inline extensions/grid/init.lua]: should probably remove margins here, as they are meaningful when talking about window frames but I don't they should apply to the "abstract" grid cell

File: extensions/grid/init.lua [hunk 3]
```diff
@@ -222,6 +222,35 @@ function grid.get(win)
...
+--- hs.grid.getCell(cell, screen) -> hs.geometry
+--- Function
+--- Gets the `hs.geometry` rect for a cell on a particular screen
+---
+--- Parameters:
+---  * cell - a cell object, i.e. an `hs.geometry` rect or argument to construct one
+---  * screen - an `hs.screen` object or argument to `hs.screen.find()` where the cell is located
+---
+--- Returns:
+---  * the `hs.geometry` rect for a cell on a particular screen
```
Comment (lowne) [inline extensions/grid/init.lua]: ..., or nil if the screen isn't found

**Comment by jfchevrette** [comment]
made the changes as suggested by @lowne.

This PR had gotten off my mind, sorry for the delay!

**Comment by asmagill** [comment]
Is this still relevant/wanted?

**Comment by cmsj** [comment]
@asmagill I think it is

**Pull request merged**